rts/motoko-rts/src/types.rs

```diff
@@ -802,7 +802,7 @@ pub(crate) unsafe fn block_size(address: usize) -> Words<u32> {
 
         TAG_OBJ_IND => size_of::<ObjInd>(),
 
-        TAG_ARRAY => {
+        TAG_ARRAY | TAG_ARRAY_SLICE_MIN.. => {
```

Comment:
I was also thinking about this change. However, the `block_size` should not be called during incremental mark phase. It will be used during incremental evacuation and updating phase, where the array slice information should already be restored. I will add a comment, that this is on purpose.